src/ooxml/java/org/apache/poi/xssf/streaming/reader/StreamedCell.java

```diff
@@ -0,0 +1,578 @@
+    @NotImplemented
+    public RichTextString getRichStringCellValue() {
+        throw new UnsupportedOperationException("Not implememted yet.");
+    }
+
+    /**
+     * <pre>
+     *  Returns the String value of cell content
+     * </pre>
+     * 
+     * @return String, representing the value of cell
+     */
+    @Override
+    public String getStringCellValue() {
+        return value;
+    }
+
+    /**
+     * <pre>
+     * Not supported right now, as StreamedWorkbook
+     * supports only reading.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     * 
+     */
+    @Override
+    @NotImplemented
+    public void setCellValue(boolean value) {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Not supported right now, as StreamedWorkbook
+     * supports only reading.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public void setCellErrorValue(byte value) {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Not supported due to memory foot print.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public boolean getBooleanCellValue() {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Will be supported in future.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public byte getErrorCellValue() {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Not supported right now, as StreamedWorkbook
+     * supports only reading.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public void setCellStyle(CellStyle style) {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Will be supported in future.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public CellStyle getCellStyle() {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Not supported right now, as StreamedWorkbook
+     * supports only reading.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public void setAsActiveCell() {
+        throw new UnsupportedOperationException("Operation not supported.");
+    }
+
+    /**
+     * <pre>
+     * Will be supported in future.
+     * </pre>
+     * 
+     * @exception UnsupportedOperationException
+     */
+    @Override
+    @NotImplemented
+    public CellAddress getAddress() {

```

Comment:
Thanks Javen for the review. So far, I've only really just taken the original patch and got the existing tests working again with the latest POI code.
I've just fixed up some of the issues you highlighted in StreamedCell. The next thing, I guess, is to look at the Cell value to ensure it has full type support and not just supporting Strings.